src/test/java/org/numenta/nupic/encoders/RandomDistributedScalarEncoderTest.java

```diff
@@ -0,0 +1,422 @@
+		assertTrue("overlapOK didn't work for far values",
+				rdse.overlapOK(48, 50, 3));
+		assertTrue("overlapOK didn't work for far values",
+				rdse.overlapOK(46, 50, 1));
+		assertTrue("overlapOK didn't work for far values",
+				rdse.overlapOK(45, 50, rdse.getMaxOverlap()));
+		assertFalse("overlapOK didn't work for far values",
+				rdse.overlapOK(48, 50, 4));
+		assertFalse("overlapOK didn't work for far values",
+				rdse.overlapOK(48, 50, 2));
+		assertFalse("overlapOK didn't work for far values",
+				rdse.overlapOK(46, 50, 2));
+		assertFalse("overlapOK didn't work for far values",
+				rdse.overlapOK(50, 50, 6));
+	}
+
+	@Test
+	public void testCountOverlap() {
+		builder = RandomDistributedScalarEncoder.builder().name("enc")
+				.resolution(1).n(500);
+		rdse = builder.build();
+
+		int[] r1 = new int[] { 1, 2, 3, 4, 5, 6 };
+		int[] r2 = new int[] { 1, 2, 3, 4, 5, 6 };
+		assertEquals("countOverlap result is incorrect", 6,
+				rdse.countOverlap(r1, r2));
+
+		r1 = new int[] { 1, 2, 3, 4, 5, 6 };
+		r2 = new int[] { 1, 2, 3, 4, 5, 7 };
+		assertEquals("countOverlap result is incorrect", 5,
+				rdse.countOverlap(r1, r2));
+
+		r1 = new int[] { 1, 2, 3, 4, 5, 6 };
+		r2 = new int[] { 6, 5, 4, 3, 2, 1 };
+		assertEquals("countOverlap result is incorrect", 6,
+				rdse.countOverlap(r1, r2));
+
+		r1 = new int[] { 1, 2, 8, 4, 5, 6 };
+		r2 = new int[] { 1, 2, 3, 4, 9, 6 };
+		assertEquals("countOverlap result is incorrect", 4,
+				rdse.countOverlap(r1, r2));
+
+		r1 = new int[] { 1, 2, 3, 4, 5, 6 };
+		r2 = new int[] { 1, 2, 3 };
+		assertEquals("countOverlap result is incorrect", 3,
+				rdse.countOverlap(r1, r2));
+
+		r1 = new int[] { 7, 8, 9, 10, 11, 12 };
+		r2 = new int[] { 1, 2, 3, 4, 5, 6 };
+		assertEquals("countOverlap result is incorrect", 0,
+				rdse.countOverlap(r1, r2));
+	}
+
+	@Ignore
+	@Test
+	public void testEncodeInvalidInputType() {
+		// not valid for java as it does not follow duck typing
+	}
+
+	private List<Integer> getRangeAsList(int lowerBound, int upperBound) {
+		if (lowerBound > upperBound)
+			return null;
+
+		Integer[] arr = new Integer[upperBound - lowerBound];
+		for (int i = lowerBound; i < upperBound; i++) {
+			arr[i - lowerBound] = i;
+		}
+
+		return Arrays.asList(arr);
+	}
+
+	private boolean validateEncoder(RandomDistributedScalarEncoder encoder,
+			int subsampling) {
+		for (int i = encoder.minIndex; i <= encoder.maxIndex; i++) {
+			for (int j = i + 1; j <= encoder.maxIndex; j += subsampling) {
+				if (!encoder.overlapOK(i, j))
+					return false;
+			}
+		}
+		return true;
+	}
+
+	private int computeOverlap(int[] result1, int[] result2) {
+		if (result1.length != result2.length)
+			return Integer.MIN_VALUE;
+
+		int overlap = 0;
+		for (int i = 0; i < result1.length; i++)
+			if (result1[i] == 1 && result2[i] == 1)
+				overlap++;
+
+		return overlap;
+	}
+
+	private int getOnBits(int[] input) {
+		int onBits = 0;
+		for (int i : input)
+			onBits += i;
+		return onBits;
+	}
+}
```

Comment:
Yep, you're up early (at least here, I'm in CST)

On Wed, Jan 21, 2015 at 4:45 AM, Sean Connolly <notifications@github.com>
wrote:

> In
> src/test/java/org/numenta/nupic/encoders/RandomDistributedScalarEncoderTest.java
> <https://github.com/numenta/htm.java/pull/159#discussion_r23290796>:
>
> > +
> > +		int overlap = 0;
> > +		for (int i = 0; i < result1.length; i++)
> > +			if (result1[i] == 1 && result2[i] == 1)
> > +				overlap++;
> > +
> > +		return overlap;
> > +	}
> > +
> > +	private int getOnBits(int[] input) {
> > +		int onBits = 0;
> > +		for (int i : input)
> > +			onBits += i;
> > +		return onBits;
> > +	}
> > +}
>
> Aaah, yes, you're right you're right. Yup, this needs to be updated.. were
> tests dependent on it passing (how?)?
>
> Anyway.. back to the coffee pot with me =P
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numenta/htm.java/pull/159/files#r23290796>.
>



-- 
*We find it hard to hear what another is saying because of how loudly "who
one is", speaks...*